### PR TITLE
Ranklib Changes

### PR DESCRIPTION
**Pull request by worleydl**
Addresses issues #15 and #35 

Dependency on ranklib removed, needed code has been brought into the project with unnecessary items removed.

LegacyRanklibModelParser: Parses models using ordinals
RanklibModelParser: Parses using names

### CODE REVIEW

File: src/main/java/com/o19s/es/ltr/feature/store/StoredLtrModel.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import com.o19s.es.ltr.ranker.LtrRanker;
 import com.o19s.es.ltr.ranker.parser.LtrRankerParser;
 import com.o19s.es.ltr.ranker.parser.LtrRankerParserFactory;
+import com.o19s.es.ltr.ranker.ranklib.learning.FEATURE_TYPE;
```
Comment (nomoa) [inline src/main/java/com/o19s/es/ltr/feature/store/StoredLtrModel.java]: looks like it's unused

File: src/main/java/com/o19s/es/ltr/ranker/ranklib/learning/CoorAscent.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright [2017] Wikimedia Foundation
```
Comment (nomoa) [inline src/main/java/com/o19s/es/ltr/ranker/ranklib/learning/CoorAscent.java]: maybe we should remove this copyright notice?

**Comment by worleydl** [comment]
In regards to licensing, ranklib is BSD which supports being re-licensed under Apache, the original copyright needs to be included in the source to satisfy BSD.  I think the current headers are fine given we modified the code a bit (Named featured support and slight changes to make the code compatible with the gradle build).

**Comment by nomoa** [comment]
@worleydl thanks for the info!
This PR looks good to me, I haven't run the build myself to make sure it passes but this should be good to merge.
note that I haven't reviewed the python code carefully.
Side note: there is a comment on docs/models.json stating that ranklib is deprecated maybe we can remove that line now and add a quick note on the new type `ranklib+names`?

Thanks!

**Comment by softwaredoug** [comment]
I haven't looked at this, but one note: we found a lot of model parsing bugs with Ranklib and fixed them in RankyMcRankFace, which became the dependency we used for Elasticsearch Learning to Rank. Have any of those been incorporated here?

For example, straight Ranklib was taking ridiculous amounts of time to parse a large random forest ranker, and we reduced the time to a few seconds.

File: src/main/java/com/o19s/es/ltr/ranker/ranklib/learning/CoorAscent.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author vdang
+ * 
+ * This class implements the linear ranking model known as Coordinate Ascent. It was proposed in this paper:
+ *  D. Metzler and W.B. Croft. Linear feature-based models for information retrieval. Information Retrieval, 10(3): 257-274, 2007.
+ */
+public class CoorAscent extends Ranker {
```
Comment (ebernhardson) [inline src/main/java/com/o19s/es/ltr/ranker/ranklib/learning/CoorAscent.java]: This should be unnecessary, at ranking time coordinate ascent is just a weighted sum like the current linear model

File: src/main/java/com/o19s/es/ltr/ranker/ranklib/learning/LinearRegRank.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    public LinearRegRank(List<RankList> samples, int[] features, MetricScorer scorer)
+    {
+        super(samples, features, scorer);
+    }
+    public void init()
+    {
+        PRINTLN("Initializing... [Done]");
+    }
+
+    public double eval(DataPoint p)
```
Comment (ebernhardson) [inline src/main/java/com/o19s/es/ltr/ranker/ranklib/learning/LinearRegRank.java]: this is also a simple weighted sum, like the pre-existing linear model

**Comment by softwaredoug** [comment]
I would like to postpone this PR till after releasing 1.0 as it's rather large, and right now I think we can get by with the ranklib dependency. Any opposition?

**Comment by worleydl** [comment]
No opposition from me.  Sorry for taking a bit to get things wrapped up, I
should have free time next week.

**Comment by softwaredoug** [comment]
Delaying to 1.1...

**Pull request closed without merging**